operator-common/src/main/java/io/strimzi/operator/common/model/PemTrustSet.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.model;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.common.Util;
+
+import java.nio.charset.StandardCharsets;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the set of certificates to be trusted by a TLS client or server
+ */
+public class PemTrustSet {
+    private static final String CERT_SUFFIX = ".crt";
+    private final Map<String, byte[]> trustedCertificateMap;
+    private String secretName;
+    private String secretNamespace;
```

Comment:
Why is this needed?

Reply:
We use it for the exceptions if we fail to parse the data somehow later. It's because we lazily parse the data rather than doing it in the constructor, since I was struggling to get the tests to work if we validate upfront.